changes/views.py

```diff
@@ -310,3 +311,20 @@ def post(self, request):
         msisdn = phonenumbers.format_number(msisdn, phonenumbers.PhoneNumberFormat.E164)
         tasks.process_whatsapp_contact_check_fail.delay(str(request.user.pk), msisdn)
         return Response(status=status.HTTP_202_ACCEPTED)
+
+
+class SeedMessageSenderFailedMsisdnHook(generics.GenericAPIView):
+    """
+    Receives events from the Seed Message Sender. Supports:
+    * identity.no_address hook
+
+    Requires token auth in the querystring "token" field.
+    """
+
+    serializer_class = SeedMessageSenderFailedMsisdnHookSerializer
+    authentication_classes = (TokenAuthQueryString,)
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
```

Comment:
We need to create the `Change` object here, otherwise this view doesn't do the action, it just validates the request and then returns